storage/database/src/lib.rs

```diff
@@ -12,7 +12,17 @@
     // Rust thinks `env_inner` can be dropped earlier
     // but it cannot, we need it for the lifetime of
     // the database transaction + tables.
-    clippy::significant_drop_tightening
+    clippy::significant_drop_tightening,
+)]
+#![forbid(
```

Comment:
`cuprate_database` uses `unsafe` for:
- `pub unsafe trait DatabaseRo`
- `heed`

The `#[expect]`s everywhere may seem a bit messy although it does mean the rest of `cuprate_database` is known not to use `unsafe`, this may be worth it.